src/SmartComponents/DeletePolicy/DeletePolicy.js

```diff
@@ -107,8 +83,127 @@ const DeletePolicy = () => {
   );
 };
 
-DeletePolicy.propTypes = {
-  policy: propTypes.object,
+DeletePolicyBase.propTypes = {
+  query: propTypes.shape({
+    loading: propTypes.bool,
+    error: propTypes.object,
+    data: propTypes.object,
+  }),
+  onClose: propTypes.func,
+  deletePolicy: propTypes.func,
+};
+
+const DeletePolicyRest = ({ policyId, onClose }) => {
+  const query = usePolicyQuery2({
+    policyId,
+  });
+
+  const data = query.data?.data
+    ? dataSerialiser(query.data.data, dataMap)
+    : undefined;
+
+  const deletePolicy = (id) => {
```

Comment:
This code is tested with both API requests, failing and passing.
```suggestion
 const deletePolicy = async (id) => {
    try {
      await apiInstance.deletePolicy(id);
      dispatchAction(
        addNotification({
          variant: 'success',
          title: `Deleted "${query.data?.data?.title}" and its associated reports`,
        })
      );
      onClose();
    } catch (e) {
      dispatchAction(
        addNotification({
          variant: 'danger',
          title: 'Error removing policy',
          description: e?.message,
        })
      );
      onClose();
    }
  };
```